### PR TITLE
JSON data shown in JSON format while editing

### LINKED ISSUE #12944: Nice view for JSON data
### Steps to reproduce
- View a table with a JSON column
- Edit a row of a table with JSON data

### Expected behaviour
Have an easy view of the JSON data of a column of a table. Right now shows in only one line, without any kind of format, maybe can have an option to show when you are viewing a table with JSON data or when editing a nice JSON view (something like this: http://azimi.me/json-formatter-js/)

### Actual behaviour
Shows the JSON data in one line without any kind of formatting.

### Server configuration
**Operating system**: All

**Web server:**

**Database:**

**PHP version:** All

**phpMyAdmin version:** All

### Client configuration
**Browser:** Chrome

**Operating system:** All

### PR DESCRIPTION
**Pull request by manishbisht**
Fixes: #12944 

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

Next Goal: Use JSON.parse() to check the data is in JSON format but the data type is different.

### CODE REVIEW

File: js/makegrid.js [hunk 1]
```diff
@@ -1001,6 +1002,9 @@ function PMA_makegrid(t, enableResize, enableReorder, enableVisib, enableGridEdi
                             $editArea.removeClass('edit_area_loading');
                             if (typeof data !== 'undefined' && data.success === true) {
                                 $td.data('original_data', data.value);
+                                if($td.attr('data-type')){
```
Comment (nijel) [inline js/makegrid.js]: Shouldn't the condition here be `'json' === $cell.attr('data-type')`? I'm slightly worried of parsing anything as JSON when the data actually doesn't have to be JSON...

**Comment by manishbisht** [comment]
@nijel Yes, You are right. I have made the changes.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by arno14** [comment]
I tested the inline edit of a table having multiple json field.

I double click on the field "column1" and datas are displayed correctly in a textarea with indentation.
I double click on the field "column2"  and JSON.parse fail, it display a warning on the bottom and error is described in the javascript console.

The main problem is that in this case , the textarea is displayed with the value of "column1", 
If i did not open "column1" before, the textarea is blank.

The JSON.parse fails 
- on NULL value
- on json existing in the database (maybe due to prominent character) 

So it would be better that if the JSON.parse fail, the raw data is displayed, even unformated.

A simple try catch should be OK.
```
var initialValue = PMA_getCellValue(cell);
if($cell.attr('data-type') == 'json'){
     try{
         var value = JSON.stringify(JSON.parse(initialValue), null, 4);
    }catch(err){
         var value = initialValue;
    }                       
}
```

output of the javascript console:
```
get_scripts.js.php?scripts[]=vendor/jquery/jquery.ba-hashchange-1.3.js&scripts[]=vendor/jquery/jquery.debounce-1.0.5.js&scripts[]=menu-resizer.js&scripts[]=cross_framing_protection.js&scripts[]=rte.js&scripts[]=vendor/tracekit.js&scripts[]=error_report.js&scripts[]=config.js&scripts[]=doclinks.js&scripts[]=functions.js&v=4.8.0-dev:2066 Uncaught SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at Object.showEditCell (get_scripts.js.php?scripts[]=vendor/jquery/jquery.uitablefilter.js&scripts[]=gis_data_editor.js&scripts[]=multi_column_sort.js&scripts[]=makegrid.js&scripts[]=sql.js&call_done=1&v=4.8.0-dev&call_done=1&v=4.8.0-dev:1206)
    at startGridEditing (get_scripts.js.php?scripts[]=vendor/jquery/jquery.uitablefilter.js&scripts[]=gis_data_editor.js&scripts[]=multi_column_sort.js&scripts[]=makegrid.js&scripts[]=sql.js&call_done=1&v=4.8.0-dev&call_done=1&v=4.8.0-dev:2546)
    at HTMLTableCellElement.<anonymous> (get_scripts.js.php?scripts[]=vendor/jquery/jquery.uitablefilter.js&scripts[]=gis_data_editor.js&scripts[]=multi_column_sort.js&scripts[]=makegrid.js&scripts[]=sql.js&call_done=1&v=4.8.0-dev&call_done=1&v=4.8.0-dev:2637)
    at HTMLTableCellElement.new_func (get_scripts.js.php?scripts[]=vendor/jquery/jquery.ba-hashchange-1.3.js&scripts[]=vendor/jquery/jquery.debounce-1.0.5.js&scripts[]=menu-resizer.js&scripts[]=cross_framing_protection.js&scripts[]=rte.js&scripts[]=vendor/tracekit.js&scripts[]=error_report.js&scripts[]=config.js&scripts[]=doclinks.js&scripts[]=functions.js&v=4.8.0-dev:3287)
    at HTMLTableCellElement.dispatch (get_scripts.js.php?scripts[]=vendor/jquery/jquery.min.js&scripts[]=vendor/jquery/jquery-migrate-3.0.0.js&scripts[]=vendor/sprintf.js&scripts[]=ajax.js&scripts[]=keyhandler.js&scripts[]=vendor/jquery/jquery-ui.min.js&scripts[]=vendor/js.cookie.js&scripts[]=vendor/jquery/jquery.mousewheel.js&scripts[]=vendor/jquery/jquery.event.drag-2.2.js&scripts[]=vendor/jquery/jquery-ui-timepicker-addon.js&v=4.8.0-dev:3)
    at HTMLTableCellElement.$event.dispatch (get_scripts.js.php?scripts[]=vendor/jquery/jquery.min.js&scripts[]=vendor/jquery/jquery-migrate-3.0.0.js&scripts[]=vendor/sprintf.js&scripts[]=ajax.js&scripts[]=keyhandler.js&scripts[]=vendor/jquery/jquery-ui.min.js&scripts[]=vendor/js.cookie.js&scripts[]=vendor/jquery/jquery.mousewheel.js&scripts[]=vendor/jquery/jquery.event.drag-2.2.js&scripts[]=vendor/jquery/jquery-ui-timepicker-addon.js&v=4.8.0-dev:2511)
    at HTMLTableCellElement.q.handle (get_scripts.js.php?scripts[]=vendor/jquery/jquery.min.js&scripts[]=vendor/jquery/jquery-migrate-3.0.0.js&scripts[]=vendor/sprintf.js&scripts[]=ajax.js&scripts[]=keyhandler.js&scripts[]=vendor/jquery/jquery-ui.min.js&scripts[]=vendor/js.cookie.js&scripts[]=vendor/jquery/jquery.mousewheel.js&scripts[]=vendor/jquery/jquery.event.drag-2.2.js&scripts[]=vendor/jquery/jquery-ui-timepicker-addon.js&v=4.8.0-dev:3)
```

**Pull request merged**